lib/elixir/lib/exception.ex

```diff
@@ -1639,6 +1632,22 @@ defmodule ErlangError do
     %KeyError{key: key, term: map}
   end
 
+  def normalize({:baddot, true, key, map}, _stacktrace) do
+    %ArgumentError{
+      message:
+        "could not find key #{inspect(key)} on #{inspect(map)}. " <>
+          "When using the map.field syntax, make sure the left side of the dot is a map"
+    }
+  end
+
+  def normalize({:baddot, false, key, module}, _stacktrace) do
+    %ArgumentError{
+      message:
+        "could not find function #{inspect(key)} on #{inspect(module)}. " <>
```

Comment:
I think we generally talk about functions being *in* modules?

```suggestion
        "could not find function #{inspect(key)} in #{inspect(module)}. " <>
```